sun2000_modbus/registers.py

```diff
@@ -138,17 +138,17 @@ class BatteryEquipmentRegister(Enum):
     GridChargeCutoffSOC = Register(47088, 1, datatypes.DataType.UINT16_BE, 10, "%", AccessType.RW, None)
     # ForcibleChargeDischarge = Register(47100, 1, datatypes.DataType.UINT16_BE, 1, None, AccessType.WO, mappings.ForcibleChargeDischarge) # disabled because not readable (AccessType.WO)
     FixedChargingAndDischargingPeriods = Register(47200, 41, datatypes.DataType.MULTIDATA, None, None, AccessType.RW, None)
-    PowerOfChargeFromGrid = Register(47242, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RW, None)
-    MaximumPowerOfChargeFromGrid = Register(47244, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RW, None)
+    PowerOfChargeFromGrid = Register(47242, 2, datatypes.DataType.INT32_BE, 0.1, "W", AccessType.RW, None)
```

Comment:
This will not work as the data type of `gain` is `int`. Thus, the resulting value of gain will be 0 and calling `read` or `read_formatted` will result in a division by 0. The data type has to be changed to `float` accordingly.

Unfortunately this incident is not covered by any test.

Reply:
right, changed gain to float now. pushing commit in a bit.